outputs.tf

```diff
@@ -30,42 +30,42 @@ output "svpc_host_project_id" {
 }
 
 output "subnets_names" {
-  value       = google_compute_subnetwork.subnetwork.*.name
+  value       = [for network in google_compute_subnetwork.subnetwork : network.name] #google_compute_subnetwork.subnetwork[*].name
```

Comment:
whoops! Those should to totally have been removed! fixing!

Reply:
By the by, splat syntax is still valid in 0.12, just slightly modified:

```hcl
google_compute_subnetwork.subnetwork[*].name
```

Reply:
@aaron-lane true, do you have a preferred syntax?

Reply:
@tfhartmann well, the new splat syntax might be better in that case, to avoid the for loop.

Reply:
So I initially tried the spat syntax, and noticed tests failing. Which is why I created tests for outputs, apparently, the `splat` syntax, and the `for` syntax create a slightly different output, enough so that when I use the splat syntax  for the `subnets_ips` output,  the subnet firewall module isn't able to correctly create the resource.  Because of this, I'm going to recommend keeping with the `for` syntax so that `for_each` changes don't break  the outputs of the module.  I had thought that adding the output tests in  the secondary ranges example would have caught this, but it didn't seem to, maybe I'll add output tests to the firewall module as well. 

Here's the error that returns when using the Splat syntax 

```
  ×  gcloud: gcloud configuration (1 failed)
     ✔  Command: `gcloud compute firewall-rules describe submodule-firewall-timh-ingress-internal --project=dev-xpn-networking --format=json` exit_status should eq 0
     ✔  Command: `gcloud compute firewall-rules describe submodule-firewall-timh-ingress-internal --project=dev-xpn-networking --format=json` stderr should eq ""
     ×  Command: `gcloud compute firewall-rules describe submodule-firewall-timh-ingress-internal --project=dev-xpn-networking --format=json` internal rule should exist
     expected {"allowed" => [{"IPProtocol" => "icmp"}], "creationTimestamp" => "2019-09-26T06:53:50.896-07:00", "descript...etworking/global/firewalls/submodule-firewall-timh-ingress-internal", "sourceRanges" => ["0.0.0.0/0"]} to include {"sourceRanges" => ["10.10.20.0/24", "10.10.10.0/24"]}
     Diff:
     @@ -1,2 +1,14 @@
     -"sourceRanges" => ["10.10.20.0/24", "10.10.10.0/24"],
     +"allowed" => [{"IPProtocol"=>"icmp"}],
     +"creationTimestamp" => "2019-09-26T06:53:50.896-07:00",
     +"description" => "Allow ingress traffic from internal IP ranges",
     +"direction" => "INGRESS",
     +"disabled" => false,
     +"id" => "1123080907257904129",
     +"kind" => "compute#firewall",
     +"logConfig" => {"enable"=>false},
     +"name" => "submodule-firewall-timh-ingress-internal",
     +"network" => "https://www.googleapis.com/compute/v1/projects/dev-xpn-networking/global/networks/submodule-firewall-timh",
     +"priority" => 1000,
     +"selfLink" => "https://www.googleapis.com/compute/v1/projects/dev-xpn-networking/global/firewalls/submodule-firewall-timh-ingress-internal",
     +"sourceRanges" => ["0.0.0.0/0"],



Profile Summary: 0 successful controls, 1 control failure, 0 controls skipped
Test Summary: 2 successful, 1 failure, 0 skipped
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: 1 actions failed.
>>>>>>     Verify failed on instance <submodule-firewall-local>.  Please see .kitchen/logs/submodule-firewall-local.log for more details
>>>>>> ----------------------
>>>>>> Please see .kitchen/logs/kitchen.log for more details
>>>>>> Also try running `kitchen diagnose --all` for configuration
```

Reply:
I added some tests for outputs in the firewall submodule, and it seems that when using the splat syntax the output for `subnet_ips`  is null, and doesn't exist, with the `for` syntax, it returns a list - `["10.10.10.0/24", "10.10.20.0/24"]`  

Here's the output of tests when using the splat syntx: 
`     ↺  Input 'output_subnets_ips' does not have a value. Skipping test.` 

It looks like tests that are skipped don't result in a fail  😢

Reply:
Here are the following outputs that have no value when using the splat syntax. ( when run against the `secondary_ranges` example. 

```Shell
     ↺  Input 'output_subnets_ips' does not have a value. Skipping test.
     ↺  Input 'output_subnets_flow_logs' does not have a value. Skipping test.
     ↺  Input 'output_subnets_names' does not have a value. Skipping test.
     ↺  Input 'output_subnets_private_access' does not have a value. Skipping test.
     ↺  Input 'output_subnets_regions' does not have a value. Skipping test.
     ↺  Input 'output_subnets_secondary_ranges' does not have a value. Skipping test.
```

Reply:
Sorry for the many comments 😄  Given that the output differs, I propose we stick with the `for` syntax since it matches the current output of the module.

Reply:
Sticking with `for` sounds good to me.

Reply:
Resolving this conversation. It's my understanding that we will use the `for` syntax for these outputs.